core/src/Kokkos_View.hpp

```diff
@@ -1811,13 +1812,20 @@ KOKKOS_INLINE_FUNCTION bool operator!=(const View<LT, LP...>& lhs,
 
 namespace Kokkos {
 namespace Impl {
+struct SharedAllocationDisableTrackingGuard {
+  SharedAllocationDisableTrackingGuard() {
+    Kokkos::Impl::SharedAllocationRecord<void, void>::tracking_disable();
+  }
 
-inline void shared_allocation_tracking_disable() {
-  Kokkos::Impl::SharedAllocationRecord<void, void>::tracking_disable();
-}
+  ~SharedAllocationDisableTrackingGuard() {
+    Kokkos::Impl::SharedAllocationRecord<void, void>::tracking_enable();
+  }
+};
 
-inline void shared_allocation_tracking_enable() {
-  Kokkos::Impl::SharedAllocationRecord<void, void>::tracking_enable();
+template <class F>
+inline decltype(auto) with_shared_allocation_tracking_disabled(F&& fun) {
+  [[maybe_unused]] auto guard = SharedAllocationDisableTrackingGuard{};
+  return std::invoke(std::forward<F>(fun));
 }
```

Comment:
I would prefer something like
```suggestion
template <class FunctorType, class... Args>
inline FunctorType construct_with_shared_allocation_tracking_disabled(Args&&... args) {
  [[maybe_unused]] auto guard = SharedAllocationDisableTrackingGuard{};
  return {std::forward<Args>(args)...};
}
```

Reply:
I implemented this rather than the scope guard, because it's a bit easier to understand